service/history/mutableStateTaskGenerator.go

```diff
@@ -112,14 +117,11 @@ func newMutableStateTaskGenerator(
 }
 
 func (r *mutableStateTaskGeneratorImpl) generateWorkflowStartTasks(
+	nowTimestamp int64,
 	event *shared.HistoryEvent,
 ) error {
 
-	// TODO when refactoring workflow reset functionality,
-	//  the logic below should be changed slightly, i.e. 'now' should not be derived from
-	//  start event, but the reset time
-
-	now := time.Unix(0, event.GetTimestamp())
+	now := time.Unix(0, nowTimestamp)
```

Comment:
I remember we want to use timeSource for testiability, is it possible to do it here?